stdlib/public/core/RangeReplaceableCollection.swift.gyb

```diff
@@ -1142,11 +1142,11 @@ extension RangeReplaceableCollection
 public func + <C : RangeReplaceableCollection, S : Sequence>(lhs: C, rhs: S) -> C
   where S.Iterator.Element == C.Iterator.Element {
 
-  var lhs = lhs
```

Comment:
Without documenting it, I don't think it's OK to to let the behavior stand. We could provide a custom implementation on Array given we know that's a value type? It'd also be good to confirm if that optimization actually happens in practice, I'll try and reproduce it.